clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -2130,6 +2130,15 @@ void Generic_GCC::GCCInstallationDetector::init(
     return;
   }
 
+  // If --gcc-triple is specified use this instead of trying to
+  // auto-detect a triple.
+  if (const Arg *A =
+          Args.getLastArg(clang::driver::options::OPT_gcc_triple_EQ)) {
+    StringRef GCCTriple = A->getValue();
+    CandidateTripleAliases.clear();
+    CandidateTripleAliases.push_back(GCCTriple);
+  }
+
   // Compute the set of prefixes for our search.
```

Comment:
When `--gcc-triple=` is specified, should `--gcc-toolchain=` or configure-time `GCC_INSTALL_PREFIX` be ignored? (Shall we continue the effort to deprecate `GCC_INSTALL_PREFIX` even if `DEFAULT_SYSROOT` is to be retained as we don't have a replacement yet? https://reviews.llvm.org/D158218)

Reply:
No --gcc-toolchain= and GCC_INSTALL_PREFIX should be unaffected.  They just provide the install prefix and then --gcc-triple specifies the triple to use within that prefix.

I think GCC_INSTALL_PREFIX should be deprecated still because this functionality can be replicated using config files now.